approach.tex

```diff
@@ -59,10 +59,14 @@ \subsubsection{Dynamic Shader Generation}
 
 \subsubsection{Volume Picking}
-Picking in the context of volume rendering is the determination of which voxel user has clicked using a pointing device such as the mouse. Picking is one of most basic operations users performs to interact with a 3D scene. VTK legacy volume mappers support picking through an instance external to the mapper itself called ~\texttt{vtkVolumePicker}.  This class casts a ray into the volume and returns the point where the ray intersects an isosurface of a user specified opacity. This technique has certain limitations given that the picking class does not have enough information to correctly account for clipping, transfer functions and other parameters which define how the mapper renders internally, this reduces its reliability on the actual objects being picked. The inherent flexibility of the glsl-based implementation of the new mapper enables seamless integration with the ~\texttt{vtkHardwareSelector}'s interface, which allows a consistent selection of objects in a scene regardless of whether it is geometric or volumetric data.  Providing picking support directly within the fragment shader of the volume mapper ensures high selection accuracy even in situations where a volume intermixes with geometry in seemingly cumbersome ways or advanced features like when clipping planes are enabled (what you see is what you pick). Given the readily available picking styles supported by ~\texttt{vtkHardwareSelector} (e.g. ~\texttt{vtkAreaPicker}), it is possible to make a selection of a specific set of voxels.
+Picking, in the context of volume rendering, is the action of resolving the set of voxels a user has clicked on with the pointer. This provides the user with means to interact with the objects in a 3D scene. VTK legacy volume mappers support picking through an instance external to the mapper itself called ~\texttt{vtkVolumePicker}.  This class casts a ray into the volume and returns the point where the ray intersects an isosurface of a user specified opacity. This technique has certain limitations given that the picking class does not have enough information to correctly account for clipping, transfer functions and other parameters defining how the mapper actually renders, thus reducing its reliability on the actual objects being picked.
+
+VTK supports hardware-accelerated picking of goemetric data through the class ~\texttt{vtkHardwareSelector}, which uses a multiple-render-pass approach in order to resolve the various objects in a scene (actors in VTK parlance) and their corresponding primitives.  Each pass renders separately a different selection abstraction ( processes, actors, composite blocks and primitives), painting with a different color each of the various components in the scene.  The images rendered by each pass are downloaded from the GPU and subsequently analyzed in the class thereby resolving the correspondence of each pixel to its particular actor, primitive, etc.
```

Comment:
In my opinion that would be a bit too much detail for this section, I tried to emphasize the fact that the volume (new) mapper uses the same approach as other rendering mappers and explain that approach.  A snipped would be way to void:
```
if (accumColor.a >= maxOpacity)
{
    // Return the color provided by vtkHardwareSelector
    gl_FragData[0] = in_selectionPassColor
    return;
}
```